conanfile.py

```diff
@@ -0,0 +1,101 @@
+    author = "Google Inc."
+    license = "Apache-2.0"
+    exports_sources = ["*"]
+    generators = "cmake"
+
+    settings = "arch", "build_type", "compiler", "os"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "enable_lto": [True, False],
+        "enable_exceptions": [True, False],
+        "enable_gtest_tests": [True, False]
+    }
+    default_options = {"shared": False, "fPIC": True, "enable_lto": False, "enable_exceptions": True, "enable_gtest_tests": False}
+
+    _build_subfolder = "."
+
+    def source(self):
+        # Wrap the original CMake file to call conan_basic_setup
+        shutil.move("CMakeLists.txt", "CMakeListsOriginal.txt")
+        shutil.move(os.path.join("conan", "CMakeLists.txt"), "CMakeLists.txt")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            if self.settings.compiler == "Visual Studio" and float(self.settings.compiler.version.value) <= 12:
+                raise ConanInvalidConfiguration("{} {} does not support Visual Studio <= 12".format(self.name, self.version))
+            del self.options.fPIC
+
+        if tools.get_env("CONAN_RUN_TESTS") != "True":
+            self.output.info("Environment variable CONAN_RUN_TEST is not defined.")
+            self.output.info("Tests are disabled and the option enable_gtest_tests isn't available.")
+            del self.options.enable_gtest_tests
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Windows shared builds are not supported right now, see issue #639")
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+
+        if tools.get_env("CONAN_RUN_TESTS"):
+            cmake.definitions["BENCHMARK_ENABLE_TESTING"] = "ON"
+            cmake.definitions["BENCHMARK_ENABLE_GTEST_TESTS"] = "ON" if self.options.enable_gtest_tests else "OFF"
+        else:
+            cmake.definitions["BENCHMARK_ENABLE_TESTING"] = "OFF"
+            cmake.definitions["BENCHMARK_ENABLE_GTEST_TESTS"] = "OFF"
+
+        cmake.definitions["BENCHMARK_ENABLE_LTO"] = "ON" if self.options.enable_lto else "OFF"
+        cmake.definitions["BENCHMARK_ENABLE_EXCEPTIONS"] = "ON" if self.options.enable_exceptions else "OFF"
+
+        # See https://github.com/google/benchmark/pull/638 for Windows 32 build explanation
+        if self.settings.os != "Windows":
+            cmake.definitions["BENCHMARK_BUILD_32_BITS"] = "ON" if "64" not in str(self.settings.arch) else "OFF"
+            cmake.definitions["BENCHMARK_USE_LIBCXX"] = "ON" if (str(self.settings.compiler.libcxx) == "libc++") else "OFF"
+        else:
+            cmake.definitions["BENCHMARK_USE_LIBCXX"] = "OFF"
+
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build_requirements(self):
+        if tools.get_env("CONAN_RUN_TESTS") and self.options.enable_gtest_tests:
```

Comment:
This line looks kind of weird to me but having a look at `BENCHMARK_ENABLE_TESTING` and `BENCHMARK_ENABLE_GTEST_TESTS` it could make sense to have it separated. Anyway bot the option and env var for tests are well managed

Reply:
The `enable_gtest_tests` is only available when `CONAN_RUN_TESTS` env variable is set. So this should be fine.